pkg/cloud/services/ec2/helper_test.go

```diff
@@ -80,18 +147,59 @@ func newAWSCluster() *infrav1.AWSCluster {
 	}
 }
 
-func setupClusterScope(g *WithT) *scope.ClusterScope {
-	scheme, err := setupScheme()
-	g.Expect(err).NotTo(HaveOccurred())
+func newAWSClusterForEKS() *infrav1.AWSCluster {
```

Comment:
This is the wrong type, it needs to be `AWSManagedControlPlane`

Reply:
Thanks @richardcase , it makes sense. I have updated the PR